app/dao/users_dao.py

```diff
@@ -16,9 +21,7 @@ def save_model_user(usr, update_dict={}, pwd=None):
         usr.password = pwd
         usr.password_changed_at = datetime.utcnow()
     if update_dict:
-        if update_dict.get('id'):
-            del update_dict['id']
-            update_dict.pop('password_changed_at')
+        _remove_values_for_keys_if_present(update_dict, ['id', 'password', 'password_changed_at'])
```

Comment:
Why are we also `pop`ing the password field? Not saying it’s a bad idea, would just be good to know the reason.